app/services/ReferenceDataService.scala

```diff
@@ -32,7 +32,7 @@ class ReferenceDataServiceImpl @Inject() (connector: ReferenceDataConnector) ext
 
   def getCountryByCode(code: Option[String])(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Option[Country]] =
     code match {
-      case Some(countryCode) => getCountries().map(_.find(_.code.equals(countryCode)))
+      case Some(countryCode) => connector.getCountry(countryCode).map(Option(_))
```

Comment:
It looks like we can change the signature of this method to be `def getCountryByCode(code: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Option[Country]]`. No reason for it to take an option as far as I can see?